From 2b8d6d219b740426a957a525ea43392fefc5fd7b Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Wed, 16 Oct 2024 10:41:08 +0200
Subject: [PATCH] ETH: Make eth_handle public.

---
 libraries/Ethernet/src/ETH.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/Ethernet/src/ETH.h b/libraries/Ethernet/src/ETH.h
index 14d2d042..10ab283b 100644
--- a/libraries/Ethernet/src/ETH.h
+++ b/libraries/Ethernet/src/ETH.h
@@ -182,8 +182,8 @@ protected:
 public:
   void _onEthEvent(int32_t event_id, void *event_data);
 
-private:
   esp_eth_handle_t _eth_handle;
+private:
   uint8_t _eth_index;
   eth_phy_type_t _phy_type;
   esp_eth_netif_glue_handle_t _glue_handle;
-- 
2.47.0

