From bd46e676148d743e5f61addd7db5d2a63be53d65 Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Thu, 10 Oct 2024 16:18:04 +0200
Subject: [PATCH] dns: Enforce a minimum TTL of 30s

---
 src/core/dns.c | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/src/core/dns.c b/src/core/dns.c
index 7afeb489..2be6e374 100644
--- a/src/core/dns.c
+++ b/src/core/dns.c
@@ -1264,21 +1264,17 @@ dns_correct_response(u8_t idx)
     LWIP_DEBUGF(DNS_DEBUG, ("\n"));
   }
 
-  dns_call_found(idx, entry->ipaddr);
-
-  initial_ipaddr_cnt = entry->ipaddr_cnt;
-  for (i=0; i<initial_ipaddr_cnt; i++) {
-    if (entry->ttl[i] == 0) {
-      /* RFC 883, page 29: "Zero values are
-        interpreted to mean that the RR can only be used for the
-        transaction in progress, and should not be cached."
-        -> flush this entry now */
-      /* entry reused during callback? */
-      ip_addr_set_zero(&entry->ipaddr[i]);
-      entry->ipaddr_cnt--;
+  /* Enforce a minimum TTL of 30s to avoid excessive DNS queries in certain
+     workloads. This is not recommended in RFC 1035, but 30s should be
+     short enough for this to be safe. */
+  for (i=0; i<entry->ipaddr_cnt; i++) {
+    if (entry->ttl[i] < 30) {
+      entry->ttl[i] = 30;
     }
   }
 
+  dns_call_found(idx, entry->ipaddr);
+
   if (entry->ipaddr_cnt == 0) {
     if (entry->state == DNS_STATE_DONE) {
       entry->state = DNS_STATE_UNUSED;
-- 
2.47.0

