From 31d8443d2348a29115edefcd5617c0ff6b30cd8c Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Wed, 23 Oct 2024 17:18:06 +0200
Subject: [PATCH] fix(libsodium): add missing files to CMakeLists.txt

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6717703..8584aa2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,10 @@ set(SRC libsodium/src/libsodium)
 # Derived from libsodium/src/libsodium/Makefile.am
 # (ignoring the !MINIMAL set)
 set(srcs
+    "${SRC}/crypto_aead/aegis128l/aead_aegis128l.c"
+    "${SRC}/crypto_aead/aegis128l/aegis128l_soft.c"
+    "${SRC}/crypto_aead/aegis256/aead_aegis256.c"
+    "${SRC}/crypto_aead/aegis256/aegis256_soft.c"
     "${SRC}/crypto_aead/chacha20poly1305/aead_chacha20poly1305.c"
     "${SRC}/crypto_aead/xchacha20poly1305/aead_xchacha20poly1305.c"
     "${SRC}/crypto_auth/crypto_auth.c"
@@ -21,6 +25,7 @@ set(srcs
     "${SRC}/crypto_core/hsalsa20/core_hsalsa20.c"
     "${SRC}/crypto_core/hsalsa20/ref2/core_hsalsa20_ref2.c"
     "${SRC}/crypto_core/salsa/ref/core_salsa_ref.c"
+    "${SRC}/crypto_core/softaes/softaes.c"
     "${SRC}/crypto_generichash/blake2b/generichash_blake2.c"
     "${SRC}/crypto_generichash/blake2b/ref/blake2b-compress-avx2.c"
     "${SRC}/crypto_generichash/blake2b/ref/blake2b-compress-ref.c"
-- 
2.47.0

