From 6451a1df9f18c3ece6edd702d5e3605f0ed15f14 Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Mon, 12 Feb 2024 16:19:51 +0100
Subject: [PATCH] lwip: Increase DNS_TABLE_SIZE to 16.

---
 components/lwip/port/esp32/include/lwipopts.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/components/lwip/port/esp32/include/lwipopts.h b/components/lwip/port/esp32/include/lwipopts.h
index ef838b65a7..26685ee289 100644
--- a/components/lwip/port/esp32/include/lwipopts.h
+++ b/components/lwip/port/esp32/include/lwipopts.h
@@ -378,6 +378,8 @@ static inline uint32_t timeout_from_offered(uint32_t lease, uint32_t min)
 #define DNS_MAX_SERVERS                 CONFIG_LWIP_DNS_MAX_SERVERS
 #define DNS_FALLBACK_SERVER_INDEX       (DNS_MAX_SERVERS - 1)
 
+#define DNS_TABLE_SIZE 16
+
 #ifdef CONFIG_LWIP_FALLBACK_DNS_SERVER_SUPPORT
 #define FALLBACK_DNS_SERVER_ADDRESS(address)                           \
         do {    ip_addr_t *server_dns = address;                            \
-- 
2.43.1

