From f8cb0c5fdceac8c246696f583c9d63db6a1f7088 Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Fri, 18 Oct 2024 14:11:52 +0200
Subject: [PATCH] Use Tinkerforge/arduino-esp32.

---
 tools/config.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/config.sh b/tools/config.sh
index a027cef..19d58ea 100755
--- a/tools/config.sh
+++ b/tools/config.sh
@@ -28,7 +28,7 @@ fi
 AR_USER="espressif"
 
 # The full name of the repository
-AR_REPO="$AR_USER/arduino-esp32"
+AR_REPO="Tinkerforge/arduino-esp32"
 IDF_REPO="$AR_USER/esp-idf"
 AR_LIBS_REPO="$AR_USER/esp32-arduino-lib-builder"
 
-- 
2.47.0

