From 572b6b5adf8e58a8dd341c8dced548c36d231b9f Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Fri, 22 Sep 2023 11:41:08 +0200
Subject: [PATCH] LittleFS: Initialize all config members.

---
 libraries/LittleFS/src/LittleFS.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libraries/LittleFS/src/LittleFS.cpp b/libraries/LittleFS/src/LittleFS.cpp
index 608e0ba2..86552f14 100644
--- a/libraries/LittleFS/src/LittleFS.cpp
+++ b/libraries/LittleFS/src/LittleFS.cpp
@@ -81,7 +81,9 @@ bool LittleFSFS::begin(bool formatOnFail, const char * basePath, uint8_t maxOpen
     esp_vfs_littlefs_conf_t conf = {
       .base_path = basePath,
       .partition_label = partitionLabel_,
+      .partition = NULL,
       .format_if_mount_failed = false,
+      .read_only = false,
       .dont_mount = false
     };
 
-- 
2.42.0

