From f4b9d5e709812f5886082d9a4ad4d3593df9341c Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Thu, 10 Oct 2024 15:24:34 +0200
Subject: [PATCH] mqtt: Set ws path if not set via passed uri.

---
 mqtt_client.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/mqtt_client.c b/mqtt_client.c
index e2c7890..5bdcb17 100644
--- a/mqtt_client.c
+++ b/mqtt_client.c
@@ -613,7 +613,20 @@ esp_err_t esp_mqtt_set_config(esp_mqtt_client_handle_t client, const esp_mqtt_cl
             err = ESP_FAIL;
             goto _mqtt_set_config_failed;
         }
-    }
+    } else if (client->config->path) {
+#if MQTT_ENABLE_WS
+        esp_transport_handle_t ws_trans = esp_transport_list_get_transport(client->transport_list, "ws");
+        if (ws_trans) {
+            esp_transport_ws_set_path(ws_trans, client->config->path);
+        }
+#endif
+#if MQTT_ENABLE_WSS
+        esp_transport_handle_t wss_trans = esp_transport_list_get_transport(client->transport_list, "wss");
+        if (wss_trans) {
+            esp_transport_ws_set_path(wss_trans, client->config->path);
+        }
+#endif
+    }
     client->config->outbox_limit = config->outbox.limit;
     esp_err_t config_has_conflict = esp_mqtt_check_cfg_conflict(client->config, config);
 
-- 
2.47.0

